src/mesh/mesh-adapter.ts

```diff
@@ -14,7 +15,16 @@ export interface MeshAdapter {
     tenants: MeshTenant[],
     startDate: Date,
     endDate: Date,
+    stats: QueryStatistics,
```

Comment:
d: As you saw you must touch almost all adapters which usually hints in OOP world that this is a suboptimal design choice as it violates the "open for extension, closed for change" mantra. 
Only the src/mesh/stats-mesh-adapter-decorator.ts required this class but the other implementations don't however you now must carry over this parameter needlessly into them (actually you did not even modify the classes and I wonder how this does even compile because now they mismatch with the interface).
What I also find helpful in order to check a design is to mentally remove the feature again. If you would now delete the QueryStatistics class I would need to again change ~12 files to get rid of. With my suggested change you only would need to slightly adapt the factory, and remove the QueryStatisticsDecorator and you would be fine.